From 4135828a4b66eb7403d68839da701257c0ccd31e Mon Sep 17 00:00:00 2001
From: Xavier Leroy <xavier.leroy@college-de-france.fr>
Date: Thu, 12 Dec 2019 16:41:17 +0100
Subject: [PATCH] Avoid duplicate definitions of "common" global variables

The variables are caml_debug_info and caml_atom_table.
The multiple definitions look like a cut-and-paste error.
They cause problems with C compilers that don't follow the "common" model.

Fixes: #9144
(cherry picked from commit 53327d777ddcacf829c77c0bd3569a8ade2c72b8)
---
 asmrun/startup.c    | 1 -
 byterun/backtrace.c | 3 ---
 2 files changed, 4 deletions(-)

diff --git a/asmrun/startup.c b/asmrun/startup.c
index 070f0c64f1f2..cf8a56f35eb1 100644
--- a/asmrun/startup.c
+++ b/asmrun/startup.c
@@ -44,7 +44,6 @@
 #endif
 
 extern int caml_parser_trace;
-CAMLexport header_t caml_atom_table[256];
 char * caml_code_area_start, * caml_code_area_end;
 struct ext_table caml_code_fragments_table;
 
diff --git a/byterun/backtrace.c b/byterun/backtrace.c
index 8dfe9b7e3831..436a40722bff 100644
--- a/byterun/backtrace.c
+++ b/byterun/backtrace.c
@@ -27,9 +27,6 @@
 #include "caml/backtrace_prim.h"
 #include "caml/fail.h"
 
-/* The table of debug information fragments */
-struct ext_table caml_debug_info;
-
 CAMLexport int32_t caml_backtrace_active = 0;
 CAMLexport int32_t caml_backtrace_pos = 0;
 CAMLexport backtrace_slot * caml_backtrace_buffer = NULL;
